### DISCUSSION TITLE
The screen flashes with bad pixels when using VSync in fullscreen or using any frame limiter in borderless mode

### DISCUSSION THREAD

**Discussion by Romc3** [ppy/osu#26122] [Q&A] [answered]
### Type

Game behaviour

## Description:

Setting "Screen mode" to "Fullscreen" with "Frame limiter" set to "VSync", the screen flashes with bad pixels.

In "Borderless" mode, no matter of what frame limiter is used, the bug remains.

In "Windowed" mode everything is ok.

Recording the screen using OBS Studio cannot capture this bug.

Interestingly, when using an external 1080p 144hz monitor, the bug doesn't appears.

## My PC specs

CPU: AMD Ryzen 7 4800H with Radeon Graphics 2.90 GHz
RAM: 16 GB
Monitor: 2880x1800 90Hz (laptop's integrated display)
GPU: AMD Radeon(TM) Graphics
OS: Windows 11 Pro 22H2 22621.2861

Following instructions from AMD(prefer OEM drivers rather then manually installing AMD Software), I use driver installed from windows update 

### Screenshots or videos

https://github.com/ppy/osu/assets/76838339/79a3e85c-374f-4978-b81d-d290154f9723

![IMG_3861](https://github.com/ppy/osu/assets/76838339/03a3ca34-5796-4e76-8f5a-82aaf846d309)

### Version

osu! 2023.1224.0-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/13764700/compressed-logs.zip)

**Comment by longnguyen2004** [comment]
The recording is entirely black, and like you said, you couldn't capture it anyway.
Also, there's a common "wisdom" among PC gamers, which is to never trust OEM/Windows Update drivers (due to them usually being months, if not years outdated), and always get the latest driver from the GPU manufacturer. I've also experienced this firsthand, where switching from OEM drivers to NVIDIA drivers gave me a 1.6x performance boost in stable, and fixed a crash in lazer. So I'd recommend switching to AMD drivers first before moving on.

**Comment by Romc3** [comment]
> The recording is entirely black, and like you said, you couldn't capture it anyway. Also, there's a common "wisdom" among PC gamers, which is to never trust OEM/Windows Update drivers (due to them usually being months, if not years outdated), and always get the latest driver from the GPU manufacturer. I've also experienced this firsthand, where switching from OEM drivers to NVIDIA drivers gave me a 1.6x performance boost in stable, and fixed a crash in lazer. So I'd recommend switching to AMD drivers first before moving on.

I updated the issue using a H.264 recording, instead of HEVC. Now it should work.

Regarding drivers issue, AMD recommends OEM-provided drivers. You can see this statement in https://www.amd.com/en/support/apu/amd-ryzen-processors/amd-ryzen-7-mobile-processors-radeon-graphics/amd-ryzen-7-4800h

In fact I encountered the compatibility issue [PA-300](https://www.amd.com/en/support/kb/faq/pa-300) when using drivers installed from AMD, so I rollbacked to the OEM driver.

**Comment by longnguyen2004** [comment]
Alright, let's go with that. What's the driver version and date?

**Reply by Romc3** [reply]
The driver is downloaded from OEM's website, the version is 27.20.11028.11001 and it hasen't been updated since 2021.

I noted that when the frame limiter is set to "VSync" with fullscreen mode, according to game's internal FPS overlay, the fps is not capped to 90, while "2x" behaves as expected.

In "borderless" mode, using "VSync" still cannot cap the FPS, and furthermore, using "2x" cannot eliminate those flashes although the FPS is correct.

**Reply by longnguyen2004** [reply, marked as answer]
"hasn't been updated since 2021" doesn't give me much confidence. I have a 5600H, that also has a Vega iGPU, with latest drivers, and I don't have the same problem.

Please try the drivers from [here](https://www.amd.com/en/support/apu/amd-ryzen-processors/amd-ryzen-7-mobile-processors-radeon-graphics/amd-ryzen-7-4800h) and see if it fixes your problem.

**Reply by Romc3** [reply]
Installing AMD Software resolves the problem indeed, but another problem pops up. I guess this is not relevant to osu!lazer, I'd better ask for help in AMD's forum...

**Reply by longnguyen2004** [reply]
If you need more help, we can take this discussion to my Discord
